src/Extension/ExtendableTrait.php

```diff
@@ -181,6 +182,8 @@ public function addDynamicProperty($dynamicName, $value = null)
             $this->{$dynamicName} = $value;
         }
 
+        $this->extensionData['dynamicProperties'][$dynamicName] = $value;
```

Comment:
@joakimbo why is this necessary?

Reply:
@joakimbo if you need to store a list of properties that have been added dynamically, then store that, but don't store their values as well because those values only represent the values that they have been initialized with, it doesn't include the values that they may have been changed to. So store them as `$this->extensionData['dynamicProperties'][] = $dynamicValue;` and access them as `$this->{$dynamicName}`

Reply:
@LukeTowers Ok, will fix that. Just to be clear I had no problem accessing the properties by name, the problem arose when the Purgeable behavior needed access to all dynamicProperties without knowing their names.